### ISSUE TITLE
32kHz Vocos Multi Speaker Model Training Log 

### ISSUE THREAD

**Issue by LEECHOONGHO** [charactr-platform/vocos#48]
Training Loss, Generated Outputs.

I hope this will be a reference for model training.

https://api.wandb.ai/links/xi-speech-team/k0kdfwch

**Comment by patriotyk** [comment]
Do you have a standard tensorboard logs?  It is interesting to compare.

**Comment by LEECHOONGHO** [comment]
@patriotyk Sorry, I've change the code to log on WandB server. I have no local logging files nor tensorboard logs.

**Comment by patriotyk** [comment]
What is your validation loss on the last checkpoint? It is encoded in to the checkpoint file name. I am training 44100 for an almost a week already and loss still goes down.

**Comment by Jon-Zbw** [comment]
> Training Loss, Generated Outputs.
> 
> I hope this will be a reference for model training.
> 
> https://api.wandb.ai/links/xi-speech-team/k0kdfwch

TKS for your work,could your share 32k model training detail like:
your encodec model(i found pretrained models :24k and 48k,so i guess 32k resample to 24k or 48k for encodec pretrained model,then resample to 32k ??)

**Comment by LEECHOONGHO** [comment]
> > Training Loss, Generated Outputs.
> > I hope this will be a reference for model training.
> > https://api.wandb.ai/links/xi-speech-team/k0kdfwch
> 
> TKS for your work,could your share 32k model training detail like: your encodec model(i found pretrained models :24k and 48k,so i guess 32k resample to 24k or 48k for encodec pretrained model,then resample to 32k ??)

I'm sry for your confuse.
I just trained Mel Vocoder not for encodec's decoder.

But I have plans to train Mel-Encodec?(Mel Spectrogram to RVQ Encoder, and Vocos Decoder for Various Speech data) in the future.

**Comment by LEECHOONGHO** [comment]
> Do you have a standard tensorboard logs? It is interesting to compare.



> What is your validation loss on the last checkpoint? It is encoded in to the checkpoint file name. I am training 44100 for an almost a week already and loss still goes down.

I estimated mel loss, and Generator loss with newly gained dataset. and each was 0.0942 and 2.82.
Because of the dataset's Size, estimating Eval loss with eval dataset have no difference with sampled train data.

how about your model output's quality? any artifacts?

**Comment by patriotyk** [comment]
> > Do you have a standard tensorboard logs? It is interesting to compare.
> 
> > What is your validation loss on the last checkpoint? It is encoded in to the checkpoint file name. I am training 44100 for an almost a week already and loss still goes down.
> 
> I estimated mel loss, and Generator loss with newly gained dataset. and each was 0.0942 and 2.82. Because of the dataset's Size, estimating Eval loss with eval dataset have no difference with sampled train data.
> 
> how about your model output's quality? any artifacts?

I am still training(third week). It is very slow. I will update with my results when finish.

**Comment by Mahmoud-ghareeb** [comment]
how much data do we need for training

**Comment by patriotyk** [comment]
@LEECHOONGHO  I have published my model here https://huggingface.co/patriotyk/vocos-mel-hifigan-compat-44100khz
Sounds great, and there is metrics.
@Mahmoud-ghareeb My model has been trained on 800+ hours of audio. Vocoder doesn't require text transcripts so you can easily use audio books for training. You even don't need to cut it by silence because vocos anyway internally splits provided audios to smaller segments.

**Comment by Mahmoud-ghareeb** [comment]
Great work! @patriotyk, Thank you so much

**Comment by bzp83** [comment]
> @LEECHOONGHO I have published my model here https://huggingface.co/patriotyk/vocos-mel-hifigan-compat-44100khz Sounds great, and there is metrics. @Mahmoud-ghareeb My model has been trained on 800+ hours of audio. Vocoder doesn't require text transcripts so you can easily use audio books for training. You even don't need to cut it by silence because vocos anyway internally splits provided audios to smaller segments.

I'm new to this... Could you please tell me what's the purpose of sharing the model? I mean, when I try to use it with a wav file, the output is very close to the original input file... So I'm confused here.

Thank you

**Comment by patriotyk** [comment]
This model generates audio from mel spectrograms. The functionality that you tried just generates mel from audio and then back audio from mel. But real tts systmes generate mels directly from text then vocoder generates audio.

**Comment by bzp83** [comment]
Ah ok so generating mel from audio is different from what tts systems do? Is there any code snippet that would let me test the model you trained (ans possibly others)? Thank you!

**Comment by brunopiovan** [comment]
@LEECHOONGHO did you change the model  and/or training code?

I'm struggling to achieve a good model... mainly for voices with higher frequencies.

If so, did you publish somewhere I can check?

Thanks!

**Comment by patriotyk** [comment]
You can look to my checkpoint https://huggingface.co/patriotyk/vocos-mel-hifigan-compat-44100khz there is also tensorboard logs with audio and config. But it is trained slightly in non standard way. It is trained to accept standard hifigan 22.5 khz mels and outputs 44.1 khz audio. Also there is discriminators you can just finetune it for your domain.

**Comment by brunopiovan** [comment]
> You can look to my checkpoint https://huggingface.co/patriotyk/vocos-mel-hifigan-compat-44100khz there is also tensorboard logs with audio and config. But it is trained slightly in non standard way. It is trained to accept standard hifigan 22.5 khz mels and outputs 44.1 khz audio. Also there is discriminators you can just finetune it for your domain.

Thank you! I'll check it out!

So it does super resolution? Can you share more details on how you trained/configured it?

Also, does it matter to have MelSpecReconstructionLoss config different from feature_extractor? I noticed that it only changes the sample rate and keep the rest of the params unchanged. Of course they're used to compute loss only, but shouldn't it use the same params to make it more consistent?

Thanks!
